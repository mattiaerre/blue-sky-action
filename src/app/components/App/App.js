import React from 'react';
import PropTypes from 'prop-types';
import { Route, Switch } from 'react-router-dom';
import Articles from '../../containers/Articles/Articles';
import Blog from '../../containers/Blog/Blog';
import Footer from '../../components/Footer/Footer';
import Home from '../../components/Home/Home';
import Kanyini from '../../components/Kanyini/Kanyini';
import Navbar from '../../components/Navbar/Navbar';
import ForSale from '../../components/ForSale/ForSale';

const App = props => [
  <Navbar categories={props.categories} key="navbar" sources={props.sources} />,
  <div className="container-fluid bg-light" key="container">
    <Switch>
      <Route component={Home} exact path="/" />
      <Route component={Blog} exact path="/blog" />
      <Route component={Blog} exact path="/blog/:slug" />
      <Route component={Kanyini} exact path="/kanyini" />
      <Route component={ForSale} exact path={props.forSalePath} />
      <Route
        exact
        path="/:category/:source"
        render={_ => (
          <Articles {..._} articles={props.articles} sources={props.sources} />
        )}
      />
    </Switch>
  </div>,
  <Footer key="footer" name={props.name} version={props.version} />
];

App.propTypes = {
  articles: PropTypes.array.isRequired,
  categories: PropTypes.array.isRequired,
  forSalePath: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  sources: PropTypes.array.isRequired,
  version: PropTypes.string.isRequired,
  weather: PropTypes.array.isRequired
};

export default App;
