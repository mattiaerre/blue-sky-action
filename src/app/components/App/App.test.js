import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import App from './App';

describe('<App />', () => {
  const nullWeather = {
    id: 0,
    main: {},
    sys: {},
    weather: [{ icon: '' }]
  };

  fetch.mockResponse(
    JSON.stringify([
      {
        description:
          'Your daily roundup of the biggest TechCrunch stories and startup news.',
        publishedAt: '2017-10-31T20:00:33Z',
        title: 'Crunch Report',
        url: 'https://techcrunch.com/video/crunchreport/'
      },
      {
        description:
          "It's impossible to know just how much stuff being circulated on social networks is Russian state content in sheep's clothing, although tech companies are..",
        publishedAt: '2017-11-01T23:49:18Z',
        title:
          'Hereâ€™s how Russia targeted its fake Facebook ads and how those ads performed',
        url:
          'https://techcrunch.com/2017/11/01/list-russian-ads-facebook-instagram/'
      }
    ])
  );

  const props = {
    articles: [],
    categories: [],
    name: '',
    sources: [
      {
        category: 'entertainment',
        id: 'buzzfeed',
        name: 'Buzzfeed'
      }
    ],
    version: '',
    weather: { home: [nullWeather], kanyini: [nullWeather] }
  };

  const scenarios = [
    {
      location: '/'
    },
    {
      location: '/kanyini'
    },
    {
      location: '/entertainment'
    },
    {
      location: '/entertainment/buzzfeed'
    }
  ];

  scenarios.forEach(({ location }) => {
    describe('location === "/"', () => {
      test('to match snapshot', () => {
        const tree = create(
          <StaticRouter location={location} context={{}}>
            <App {...props} />
          </StaticRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
