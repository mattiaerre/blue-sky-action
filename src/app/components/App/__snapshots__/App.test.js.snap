// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> location === "/" to match snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light border border-top-0 border-right-0 border-left-0"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      [\`/\`]
    </a>
    <button
      aria-controls="navbar"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbar"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbar"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="true"
            className="nav-link text-light active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={false}
            className="nav-link text-light"
            href="/kanyini"
            onClick={[Function]}
            style={undefined}
          >
            Kanyini
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container-fluid bg-light"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <h2>
            <svg
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 40 40"
              width="1em"
            >
              <g>
                <path
                  d="m20 5s-10.3 8.9-16.1 13.7c-0.3 0.3-0.6 0.8-0.6 1.3 0 0.9 0.8 1.7 1.7 1.7h3.3v11.6c0 1 0.8 1.7 1.7 1.7h5c0.9 0 1.7-0.7 1.7-1.7v-6.6h6.6v6.6c0 1 0.8 1.7 1.7 1.7h5c0.9 0 1.7-0.7 1.7-1.7v-11.6h3.3c0.9 0 1.7-0.8 1.7-1.7 0-0.5-0.3-1-0.7-1.3-5.7-4.8-16-13.7-16-13.7z"
                />
              </g>
            </svg>
             
            <span
              className="align-middle"
            >
              Home
            </span>
          </h2>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <h3>
            Loading ...
          </h3>
        </div>
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-12"
          >
            <h3>
              Interesting links
            </h3>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-12"
          >
            <ul
              className="list-unstyled"
            >
              <li>
                <a
                  href="https://www.airnow.gov/index.cfm?action=airnow.local_city&cityid=317"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  AirNow &gt; California &gt; San Francisco
                </a>
              </li>
              <li>
                <a
                  href="http://rammb-slider.cira.colostate.edu/?sat=goes-16&sec=full_disk&x=5611.5&y=3628&z=5&im=18&ts=1&st=0&et=0&speed=130&motion=loop&map=0&lat=0&p%5B0%5D=16&opacity%5B0%5D=1&hidden%5B0%5D=0&pause=0&slider=-1&hide_controls=0&mouse_draw=0&s=rammb-slider"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Satellite View
                </a>
              </li>
              <li>
                <a
                  href="https://www.youtube.com/watch?v=CMXiCR2gQw0"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Above & Beyond #ABGT250 Live at The Gorge Amphitheatre, Washington State (Full 4K HD Set)
                </a>
              </li>
              <li>
                <a
                  href="https://vietnam.visahq.com/requirements/italy/resident-united-states/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  VisaHQ
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="border border-right-0 border-bottom-0 border-left-0"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <small
            className="float-right"
          >
             v
          </small>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <small
            className="float-right"
          >
            <a
              href="https://newsapi.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              powered by News API
            </a>
          </small>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`<App /> location === "/entertainment/buzzfeed" to match snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light border border-top-0 border-right-0 border-left-0"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      [\`/\`]
    </a>
    <button
      aria-controls="navbar"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbar"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbar"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="true"
            className="nav-link text-light active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={false}
            className="nav-link text-light"
            href="/kanyini"
            onClick={[Function]}
            style={undefined}
          >
            Kanyini
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container-fluid bg-light"
  >
    <div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-12"
          >
            <h2>
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m35 6.7h-30c-1.8 0-3.3 1.5-3.3 3.3v20c0 1.8 1.5 3.3 3.3 3.3h30c1.8 0 3.3-1.5 3.3-3.3v-20c0-1.8-1.5-3.3-3.3-3.3z m-30 3.3h13.3v20h-13.3v-20z m30 20h-15v-20h15l0 20z m-1.7-7.5c0-0.5-0.3-0.8-0.8-0.8h-1.7c-0.4 0-0.8 0.3-0.8 0.8v5c0 0.5 0.4 0.8 0.8 0.8h1.7c0.5 0 0.8-0.3 0.8-0.8v-5z m-5-10c0-0.5-0.3-0.8-0.8-0.8h-5c-0.5 0-0.8 0.3-0.8 0.8v8.3c0 0.5 0.3 0.9 0.8 0.9h5c0.5 0 0.8-0.4 0.8-0.9v-8.3z m2.5 4.2h1.7c0.5 0 0.8-0.4 0.8-0.9s-0.3-0.8-0.8-0.8h-1.7c-0.4 0-0.8 0.4-0.8 0.8s0.4 0.9 0.8 0.9z m0 3.3h1.7c0.5 0 0.8-0.4 0.8-0.8s-0.3-0.9-0.8-0.9h-1.7c-0.4 0-0.8 0.4-0.8 0.9s0.4 0.8 0.8 0.8z m-8.3 5h5c0.5 0 0.8-0.4 0.8-0.8s-0.3-0.9-0.8-0.9h-5c-0.5 0-0.8 0.4-0.8 0.9s0.3 0.8 0.8 0.8z m5 1.7h-5c-0.5 0-0.8 0.3-0.8 0.8s0.3 0.8 0.8 0.8h5c0.5 0 0.8-0.3 0.8-0.8s-0.3-0.8-0.8-0.8z m3.3-13.4h1.7c0.5 0 0.8-0.3 0.8-0.8s-0.3-0.8-0.8-0.8h-1.7c-0.4 0-0.8 0.3-0.8 0.8s0.4 0.8 0.8 0.8z m-14.1-0.8c0-0.5-0.4-0.8-0.9-0.8h-8.3c-0.5 0-0.8 0.3-0.8 0.8v5c0 0.5 0.3 0.8 0.8 0.8h8.3c0.5 0 0.9-0.3 0.9-0.8v-5z m-0.9 10.8h-8.3c-0.5 0-0.8 0.4-0.8 0.9s0.3 0.8 0.8 0.8h8.3c0.5 0 0.9-0.4 0.9-0.8s-0.4-0.9-0.9-0.9z m0-3.3h-8.3c-0.5 0-0.8 0.4-0.8 0.8s0.3 0.9 0.8 0.9h8.3c0.5 0 0.9-0.4 0.9-0.9s-0.4-0.8-0.9-0.8z m0 6.7h-8.3c-0.5 0-0.8 0.3-0.8 0.8s0.3 0.8 0.8 0.8h8.3c0.5 0 0.9-0.3 0.9-0.8s-0.4-0.8-0.9-0.8z"
                  />
                </g>
              </svg>
               
              <span
                className="align-middle"
              >
                Entertainment
              </span>
            </h2>
          </div>
        </div>
        <div
          className="row pb-3"
        >
          <div
            className="col-12"
          >
            <ul
              className="nav nav-pills"
            >
              <li
                className="nav-item"
              >
                <a
                  aria-current="true"
                  className="nav-link active"
                  href="/entertainment/buzzfeed"
                  onClick={[Function]}
                  style={Object {}}
                >
                  Buzzfeed
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <h3>
            Loading ...
          </h3>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="border border-right-0 border-bottom-0 border-left-0"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <small
            className="float-right"
          >
             v
          </small>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <small
            className="float-right"
          >
            <a
              href="https://newsapi.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              powered by News API
            </a>
          </small>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`<App /> location === "/for-sale" to match snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light border border-top-0 border-right-0 border-left-0"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      [\`/\`]
    </a>
    <button
      aria-controls="navbar"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbar"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbar"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="true"
            className="nav-link text-light active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={false}
            className="nav-link text-light"
            href="/kanyini"
            onClick={[Function]}
            style={undefined}
          >
            Kanyini
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container-fluid bg-light"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <h2>
            For Sale
          </h2>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card"
          >
            <img
              alt=""
              className="card-img-top"
              src="/images/IMG_0235.JPG"
            />
            <div
              className="card-body"
            >
              <h3
                className="card-title"
              >
                Garmin
              </h3>
              <h4>
                Forerunner 235
              </h4>
              <p
                className="card-text"
              >
                01 JAN 2017
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="border border-right-0 border-bottom-0 border-left-0"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <small
            className="float-right"
          >
             v
          </small>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <small
            className="float-right"
          >
            <a
              href="https://newsapi.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              powered by News API
            </a>
          </small>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`<App /> location === "/kanyini" to match snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light border border-top-0 border-right-0 border-left-0"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      [\`/\`]
    </a>
    <button
      aria-controls="navbar"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbar"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbar"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="true"
            className="nav-link text-light active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current="true"
            className="nav-link text-light active"
            href="/kanyini"
            onClick={[Function]}
            style={Object {}}
          >
            Kanyini
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container-fluid bg-light"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <h2>
            <svg
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 40 40"
              width="1em"
            >
              <g>
                <path
                  d="m18.3 33.3h-6.6c-1 0-1.7 0.8-1.7 1.7s0.7 1.7 1.7 1.7h16.6c1 0 1.7-0.8 1.7-1.7s-0.7-1.7-1.7-1.7h-6.6v-2c2.7-0.6 5.2-2 7.2-4 2.9-2.9 4.4-6.6 4.4-10.6 0-3.5-1.1-6.7-3.2-9.4l1.1-1.1c0.6-0.7 0.6-1.7 0-2.4-0.7-0.6-1.7-0.6-2.4 0l-3.4 3.4 1.2 1.2c2.2 2.2 3.4 5.2 3.4 8.3s-1.2 6-3.4 8.2-5.1 3.4-8.3 3.4-6-1.2-8.2-3.4c-0.7-0.6-1.7-0.6-2.4 0-0.6 0.7-0.6 1.7 0 2.4 2.9 2.8 6.6 4.4 10.6 4.4v1.6z m0-26.6c2.8 0 5.3 1.1 7.1 2.9 1.8 1.8 2.9 4.3 2.9 7.1 0 2.7-1.1 5.2-2.9 7-1.8 1.8-4.3 3-7.1 3-2.7 0-5.2-1.2-7-3-1.8-1.8-3-4.3-3-7s1.2-5.3 3-7.1c1.8-1.8 4.3-2.9 7-2.9z"
                />
              </g>
            </svg>
             
            <span
              className="align-middle"
            >
              Kanyini
            </span>
             
            <small
              className="text-muted align-middle"
            >
              in 3 months
            </small>
          </h2>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <p
            className="m-2"
          >
            <a
              href="https://remoteyear.com/itineraries/kanyini"
              rel="noopener noreferrer"
              target="_blank"
            >
              Itinerary
            </a>
          </p>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <h3>
            Loading ...
          </h3>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="border border-right-0 border-bottom-0 border-left-0"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <small
            className="float-right"
          >
             v
          </small>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <small
            className="float-right"
          >
            <a
              href="https://newsapi.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              powered by News API
            </a>
          </small>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
